community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -142,7 +143,7 @@ TransactionWriteState upgradeToSchemaWrites() throws InvalidTransactionTypeKerne
     private TransactionHooks.TransactionHooksState hooksState;
     private final KernelStatement currentStatement;
     private final StorageStatement storageStatement;
-    private CloseListener closeListener;
+    private final List<CloseListener> closeListeners = new ArrayList<>( 2 );
```

Comment:
Refactoring is trivial. I've made it here: https://github.com/neo4j/neo4j/pull/7526.
Please look through the changes there and make use of them in this PR.